rpc/http.go

```diff
@@ -175,6 +166,28 @@ func (srv *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	srv.ServeSingleRequest(codec, OptionMethodInvocation)
 }
 
+// Returns a non-zero response code and error message if the request is invalid.
+func httpErrorResponse(r *http.Request) (int, string) {
```

Comment:
Instead of returning `(int, string)`, please return `(int, error)`. That makes it a lot cleaner imho, since you don't need to check for `0` equality outside (which is arbitrary), rather can do a clean `error` check which is the standard in Go.